### PR TITLE
Issue 14655: Only check on baseEntries after config processed

### LINKED ISSUE #14655: Move participatingBaseEntry check to avoid inaccurate logging of CWIMK0004E message
After configuration updates for a `federatedRepository` and user registries (such as a BasicRegistry or LdapRegistry), an incorrect `CWIMK0004E` message is logged indicating that the participating base entry cannot be found even though the registries are defined in the server.xml

For example:
`CWIMK0004E: The user registry operation could not be completed. The participating base entry definition o=SampleBasicRealm is not valid. Correct the participating base entry definition in the server.xml file. For example the syntax is <participatingBaseEntry name='...'/>.` message can be logged incorrectly. 

In some timing situations, the `participatingBaseEntry` list on the `federatedRepository` configuration is checked prior to loading all of the configured user registries and the `CWIMK0004E` message is logged. A fix was made to ensure checking is done after registries are loaded.

### PR DESCRIPTION
**Pull request by kristip17**
Fixes #14655 

The previous behavior was that the ConfigManager sent out a notification to Listeners only after receiving a Modify. When VMMService (the only implemented listener) received the event, it checks that the participatingBaseEntries listed match a registry. If a registry isn't found, an error message is issued (`CWIMK0004E`).

The startup timing has changed such that sometimes we process the config as an empty initialize then call modify when we may or may not have finished processing the registries (such as BasicRegistry or LdapRegistry). This caused a false negative on the error messages.

To match previous behavior and avoid the baseEntry check happening too early. I did the following:
- Add a listener for checking when server config modifications are complete
- Toggle when we're past init phase and modify is being called (otherwise, `URAPIs_RealmPropertyMappingTest` failed because we received a config complete check before initializing the CustomRegistry/CustomRepository features in that test -- I think this matches more closely to the original behavior).
- Added a call to deregister in VMMService or we ended up with a new and old VMMService to call from ConfigManager
- Added a new FAT test that recreates the failing scenario and intentional creates a scenario that should log the `CWIMK0004E`

### CODE REVIEW

**Comment by kristip17** [comment]
#build

**Comment by kristip17** [comment]
There was one failure in the testing -- it was a "too much trace" error. I pulled the logs and reviewed. While it is a security test, it wasn't logging any of the new (small) trace that I added. I also searched and found an existing bug where this test is popping too much trace, so I didn't push it over.

**Comment by kristip17** [comment]
#build

File: dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java [hunk 1]
```diff
@@ -424,10 +434,14 @@ public void registerRealmConfigChangeListener(RealmConfigChangeListener listener
             listeners.add(listener);
     }
 
+    public void deregisterRealmConfigChangeListener(RealmConfigChangeListener listener) {
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java:437]: Add javadoc.

File: dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java [hunk 2]
```diff
@@ -496,4 +510,56 @@ public long getPageCacheTimeOut() {
     void setRepositoryManager(RepositoryManager repositoryManager) {
         this.repositoryManager = repositoryManager;
     }
+
+    @Trivial
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java:514]: Remove trivial? Add javadoc.

File: dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java [hunk 3]
```diff
@@ -496,4 +510,56 @@ public long getPageCacheTimeOut() {
...
+    @Reference(service = FutureMonitor.class)
+    protected void setFutureMonitor(FutureMonitor futureMonitor) {
+        _futureMonitor = futureMonitor;
+    }
+
+    @Trivial
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java:520]: Ditto.

File: dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java [hunk 4]
```diff
@@ -496,4 +510,56 @@ public long getPageCacheTimeOut() {
...
+    /**
+     * After a configuration modification, run a RealConfigChange check which will verify the participatingBaseEntries.
+     *
+     * We used to do this directly from the modify command, but timing change during startup/modify and we would process the check
+     * before the user registries activated, leading to misleading error messages being printed.
+     */
+    @Trivial
+    @Override
+    public void notificationCreated(RuntimeUpdateManager updateManager, RuntimeUpdateNotification notification) {
+        if (modified && notification.getName().equals(RuntimeUpdateNotification.CONFIG_UPDATES_DELIVERED)) {
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java:534]: Reverse the string equality check so the constant is first to protect against possible NPE.

**Review by jvanhill — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java (hunk 1 above)
Comment (kristip17) [reply dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java:437]: Done

File: dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java (hunk 3 above)
Comment (kristip17) [reply dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java:520]: Done

File: dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java (hunk 2 above)
Comment (kristip17) [reply dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java:514]: Done

File: dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java (hunk 4 above)
Comment (kristip17) [reply dev/com.ibm.ws.security.wim.core/src/com/ibm/ws/security/wim/ConfigManager.java:534]: Done

**Comment by kristip17** [comment]
#build

File: dev/com.ibm.ws.security.wim.core_fat/publish/servers/com.ibm.ws.security.wim.core.fat.mod/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+/dropins
```
Comment (jvanhill) [inline dev/com.ibm.ws.security.wim.core_fat/publish/servers/com.ibm.ws.security.wim.core.fat.mod/.gitignore:1]: We should be building in the autofvt build directory, so not sure we should need this.

**Review by jvanhill — APPROVED** [review]
There is one comment about a gitignore file that I am not sure you need. You can delete it if it is not needed and you shouldn't have to rebuild since it would not effect the build.

File: dev/com.ibm.ws.security.wim.core_fat/publish/servers/com.ibm.ws.security.wim.core.fat.mod/.gitignore (hunk 5 above)
Comment (kristip17) [reply dev/com.ibm.ws.security.wim.core_fat/publish/servers/com.ibm.ws.security.wim.core.fat.mod/.gitignore]: Removed

**Comment by kristip17** [comment]
#libby

**Pull request merged**